codegenerator/cli/npm/envio/local-bin.mjs

```diff
@@ -4,21 +4,69 @@
 
 import { spawnSync } from "child_process";
 import path from "path";
-import { fileURLToPath } from "url";
 
 /**
  * Runs `envio` with args using nodejs spawn
  */
 function runLocalEnvio() {
   const args = process.argv.slice(2);
-  const __filename = fileURLToPath(import.meta.url); // get the resolved path to the file
-  const __dirname = path.dirname(__filename); // get the name of the directory
+
+  const caller = process.argv[1];
+  if (!caller.endsWith("/node_modules/envio/local-bin.mjs")) {
+    throw new Error(
+      `Unexpected call to local envio package. Either use cargo or use the local package via npm scripts. Caller: ${caller}`
+    );
+  }
+
+  let root = path.join(caller, "../../..");
+
+  const pnpmListResult = spawnSync("pnpm", ["list", "envio", "--json"], {
+    cwd: root,
+  });
+  if (pnpmListResult.status !== 0) {
+    throw new Error(
+      `Failed to run pnpm list envio --json. Error: ${pnpmListResult.stderr}`
+    );
+  }
+  let pnpmList;
+  try {
+    const outputString = pnpmListResult.output.toString();
+    // It starts and ends with , so we need to remove it before parsing
+    const jsonString = outputString.slice(1, -1);
+    pnpmList = JSON.parse(jsonString);
+  } catch (e) {
+    throw new Error(
+      `Invalid pnpm list envio --json output. Error: ${e.message}`
+    );
+  }
+
+  let envioVersion;
+  try {
+    envioVersion = pnpmList[0].dependencies["envio"].version;
+  } catch (e) {
+    throw new Error(
+      `Failed to get envio version from pnpm list envio --json output. Error: ${e.message}. Pnpm list: ${pnpmList}`
+    );
+  }
+
+  if (!envioVersion.startsWith("file:")) {
+    throw new Error(
+      `Unexpected envio version. It should have the file: protocol. Actual: ${envioVersion}`
```

Comment:
Perhaps this can contain a link or just an example? I would be a little confused getting this error. Some people might already have run the init command and have an existing project. I have a few for eg.

Reply:
Honestly I didn't thought about the error handling deeply, left it to Supermaven generate most of them. I think the error is fine, since if you have the `"example"` version, then you won't run the binary in any way. If you have a simple path, the you'll get the `It should have the file: protocol`.